---
title: "Preliminary analysis of 11 RNA seq datasets for novel insights"
author:
- name: Miguel Alberto Pantoja-Zepeda
  affiliation: First Author's Affiliation
- name: Second
  affiliation: Second Author's Affiliation
  email: corresponding@author.com
package: packageName
output:
  BiocStyle::html_document
abstract: |
  Description of your vignette
vignette: |
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---


# Introduction

The aim of this study was to conduct a preliminary analysis of 11 RNA seq datasets to identify potential novel insights in gene expression. By combining multiple datasets, we aimed to increase the sample size and statistical power, and thus increase our ability to identify meaningful patterns in the data.

# Methods

We obtained 11 publicly available RNA seq datasets, each of which contained gene expression data from a different biological system. We preprocessed and normalized the data using standard procedures and then combined the datasets to generate a single expression matrix. We performed unsupervised clustering to identify any subgroups within the data and used differential expression analysis to identify genes that were significantly differentially expressed between the subgroups.

# Results

Our preliminary analysis identified several potential novel insights in gene expression across the 11 datasets. We identified several subgroups within the data, which appeared to be driven by the differential expression of specific genes. In particular, we found that several genes involved in metabolic processes were differentially expressed across the datasets. We also identified several genes that were consistently upregulated or downregulated across the datasets, suggesting a potential role in regulating gene expression.

# Conclusion

Our preliminary analysis of 11 RNA seq datasets has identified several potential novel insights in gene expression. Further analysis will be necessary to validate these findings and explore their potential functional significance. However, our results demonstrate the potential of combining multiple datasets to increase statistical power and identify meaningful patterns in gene expression data.
# Getting started

To enable the _Bioconductor_ style in your R Markdown vignette you need to include the following in the `DESCRIPTION` file:

    VignetteBuilder: knitr
    Suggests: BiocStyle, knitr, rmarkdown
# Style macros

_BiocStyle_ introduces the following macros for referring to _R_ packages:

* `r Biocpkg("recount3")`, coso
* `r Biocpkg("limma")`, etc



# Style macros

_BiocStyle_ introduces the following macros for referring to _R_ packages:

* `r Biocpkg("IRanges")`, for _Bioconductor_ software, annotation and experiment data packages,
* `r CRANpkg("data.table")`, for _R_ packages available on CRAN,
* `r Githubpkg("rstudio/rmarkdown")`, for _R_ packages available on GitHub,
* `r Rpackage("MyPkg")`, for _R_ packages that are _not_ available on _Bioconductor_, CRAN or GitHub.

# Code
```{r, message=FALSE}


library(recount3)

options(recount3_url = "https://recount-opendata.s3.amazonaws.com/recount3/release") 

human_projects <- available_projects()
```


```{r,  message = FALSE}


crear_objetos_rse <- function(x) {
  rsa_gene <- create_rse_manual(
    project = x,
    project_home = "data_sources/sra",
    organism = 'mouse',
    annotation = 'gencode_v23',
    type = 'gene'
  )
  rsa_gene
}

SRA_list <- c('SRP215380',
'SRP128050',
'SRP199237',
'SRP152324',
'SRP152966',
'SRP131336',
'SRP130908',
'SRP100500',
'SRP125754',
'SRP174174',
'SRP099692'
)
#$SRP178253,SRP093683, 'SRP223471' replicas,#'SRP188879',
#'SRP119845' #'SRP130908',

#SRP130908 male/female unknown (and it shows..)

#, rse_gene_list$SRP119845, , rse_gene_list$SRP188879

rse_gene_list <- lapply(SRA_list, crear_objetos_rse)
names(rse_gene_list) <- SRA_list

```

# Explore information

```{r}

assay_compute <- function(x) {
  assay(x, "counts") <- compute_read_counts(x)
  x
}

rse_gene_list <- lapply(rse_gene_list, assay_compute)

```


#
```{r}

grep_name_cols <- function(x){
  colData(x)[
      ,
      grepl("^sra_attribute", colnames(colData(x)))
  ]
}

rse_gene_list <- lapply(rse_gene_list, expand_sra_attributes)

lapply(rse_gene_list, grep_name_cols)


```

```{r}
raw_Data <- rse_gene_list

save(raw_Data, file ="~/Miguel_Github/MERS_CoV_analysis/processed_data/raw_studies.RData")

```



```{r}
rse_gene_list <- raw_Data

rse_gene_list$SRP131336$sra_attribute.source_name_1 <- rse_gene_list$SRP131336$sra_attribute.source_name
colnames(colData(rse_gene_list$SRP131336))[colnames(colData(rse_gene_list$SRP131336)) == "sra_attribute.source_name_1"] <- "sra_attribute.Sex"



colnames(colData(rse_gene_list$SRP125754))[colnames(colData(rse_gene_list$SRP125754)) == "sra_attribute.agent"] <- "sra_attribute.genotype"

colData(rse_gene_list$SRP100500) <- subset(colData(rse_gene_list$SRP100500), select = -c(sra_attribute.tissue))  
colnames(colData(rse_gene_list$SRP100500))[colnames(colData(rse_gene_list$SRP100500)) == "sra_attribute.cell_type"] <- "sra_attribute.tissue"


rse_gene_list$SRP131336$sra_attribute.Sex <-  gsub('wild type_', '',rse_gene_list$SRP131336$sra_attribute.Sex )
rse_gene_list$SRP131336$sra_attribute.Sex <-  gsub('mutant_', '',rse_gene_list$SRP131336$sra_attribute.Sex )
rse_gene_list$SRP131336$sra_attribute.Sex <-  gsub('_brain', '',rse_gene_list$SRP131336$sra_attribute.Sex )

#rse_gene_list$SRP131336$sra_attribute.Sex


#SRP119845,SRP100500,SRP130908,SRP125754,SRP174174,SRP099692

add_males <- function(x){
  bool_1 <- sum(colnames(colData(x))=='sra_attribute.Sex')
  bool_2 <- sum(colnames(colData(x))=='sra_attribute.sex')
  bool_3 <- sum(colnames(colData(x))=='sra_attribute.gender')
  if (!(bool_1|bool_2|bool_3)){
    x$sra_attribute.Sex <- rep('unknown',dim(x)[2])}
  x
  }

add_age <- function(x){
  bool_1 <- sum(colnames(colData(x))=='sra_attribute.age')
  bool_2 <- sum(colnames(colData(x))=='sra_attribute.weeks')
  bool_3 <- sum(colnames(colData(x))=='sra_attribute.developmental_stage')
  if (!(bool_1|bool_2|bool_3)){
    x$sra_attribute.age <- rep('8 weeks',dim(x)[2])}
  x
}


rse_gene_list <- lapply(rse_gene_list, add_males)
rse_gene_list <- lapply(rse_gene_list, add_age)

colData(rse_gene_list$SRP099692) <- subset(colData(rse_gene_list$SRP099692), select = -c(sra_attribute.treatment))  
colData(rse_gene_list$SRP100500) <- subset(colData(rse_gene_list$SRP100500), select = -c(sra_attribute.tumor_status))  

rse_gene_list$SRP174174$sra_attribute.tissue <- rse_gene_list$SRP174174$sra_attribute.source_name

```


 see each colnames
```{r}
see_col_names <- function(x){
  colnames(colData(x))[grepl('sra_attribute',colnames(colData(x)))]
}

lapply(rse_gene_list, see_col_names)

```

normalize columns (revisar si no borré columnas)
```{r}
normal_column <- function(x){
  colnames(colData(x))[colnames(colData(x)) == "sra_attribute.strain/background"] <- "sra_attribute.strain"
  colnames(colData(x))[colnames(colData(x)) == "sra_attribute.strain_background"] <- "sra_attribute.strain"
  colnames(colData(x))[colnames(colData(x)) == "sra_attribute.gender"] <- "sra_attribute.Sex"
  colnames(colData(x))[colnames(colData(x)) == "sra_attribute.tissue/cell_type"] <- "sra_attribute.tissue" 
  colnames(colData(x))[colnames(colData(x)) == "sra_attribute.genotype/variation"] <- "sra_attribute.genotype"
  colnames(colData(x))[colnames(colData(x)) == "sra_attribute.high_fat_diet"] <- "sra_attribute.genotype"

  colnames(colData(x))[colnames(colData(x)) == "sra_attribute.diet"] <- "sra_attribute.genotype"
  colnames(colData(x))[colnames(colData(x)) == "sra_attribute.developmental_stage"] <- "sra_attribute.age"
  colnames(colData(x))[colnames(colData(x)) == "sra_attribute.weeks"] <- "sra_attribute.age"

  x
}

rse_gene_list <- lapply(rse_gene_list, normal_column)
lapply(rse_gene_list, see_col_names)

```
See values of age
```{r}
check_unique <- function(x){
  print(table(x$sra_attribute.age))
  print(table(x$sra_attribute.genotype))
  print(table(x$sra_attribute.Sex))    
  #print(table(x$sra_attribute.source_name)) 
  #print(table(x$sra_attribute.strain))
  print(table(x$sra_attribute.tissue))
}
lapply(rse_gene_list, check_unique)
```


```{r, echo=FALSE}
#months <- grepl('months',rse_gene_SRP040070$sra_attribute.age)
#raw_age <- rse_gene_SRP040070
#rse_gene_SRP040070$sra_attribute.age <-  gsub(' weeks old', '',rse_gene_SRP040070$sra_attribute.age )

reshape_as <- function(x){
  #x$sra_attribute.genotype <-  gsub('wild type', 'WT', x$sra_attribute.genotype )
  x$sra_attribute.age <-  factor(tolower(x$sra_attribute.age))
  #x$sra_attribute.genotype <-  factor(tolower(x$sra_attribute.genotype))
  #x$sra_attribute.Sex <-  factor(tolower(x$sra_attribute.Sex))

  #x$sra_attribute.tissue <-  factor(tolower(x$sra_attribute.tissue))
  #x$study <- factor(ifelse(x$sra_attribute.age == 'nada', "nada", metadata(x)$project))

  #x
}
lapply(rse_gene_list, reshape_as)
#rse_gene_list <- lapply(rse_gene_list, reshape_as)
```



```{r, echo=FALSE}
## Resumen de las variables de interés
get_summary <- function(x){
summary(as.data.frame(colData(x)[
    ,
    grepl("^sra_attribute.[age|Sex|genotype|tissue]", colnames(colData(x)))
]))
}

lapply(rse_gene_list, get_summary)

```
```{r}
normalize_data <- function(x){
  x$assigned_gene_prop <- x$recount_qc.gene_fc_count_all.assigned /     x$recount_qc.gene_fc_count_all.total
    print(summary(x$assigned_gene_prop))
x
}


rse_gene_list <- lapply(rse_gene_list, normalize_data)



```

```{r}
## Guardemos nuestro objeto entero por si luego cambiamos de opinión
rse_gene_unfiltered <- rse_gene_list

```

```{r}
# no hay menores a 0.3, pero si hubiera
rse_gene_list <- rse_gene_unfiltered
## Eliminemos a muestras malas 
remove_check <- function(x){
  hist(x$assigned_gene_prop)
table(x$assigned_gene_prop < 0.3)
}

lapply(rse_gene_list, remove_check)


```

```{r,  echo=FALSE}

remove_failures <- function(x){
  x <- x[, x$assigned_gene_prop > 0.3]
  print(x)
  gene_means <- rowMeans(assay(x, "counts"))
  summary(gene_means)
  x
}

rse_gene_list <- lapply(rse_gene_list, remove_failures)

```

unimos los datos
```{r}
library("edgeR") # BiocManager::install("edgeR", update = FALSE)

joined_data <-  cbind(rse_gene_list$SRP215380,rse_gene_list$SRP128050,rse_gene_list$SRP199237,rse_gene_list$SRP152324, rse_gene_list$SRP131336, rse_gene_list$SRP152966, rse_gene_list$SRP100500, rse_gene_list$SRP125754, rse_gene_list$SRP174174, rse_gene_list$SRP099692)
#, rse_gene_list$SRP119845, , rse_gene_list$SRP188879
joined_data$prenatal <- factor(ifelse(joined_data$sra_attribute.age == 'E16.5', "prenatal", "postnatal"))


joined_data$sra_attribute.tissue <-  gsub('(brain)|(Healthy Brain control)', 'Brain', joined_data$sra_attribute.tissue, ignore.case = TRUE )
joined_data$sra_attribute.tissue <-  gsub('(GL261 tumor tissue, No treat control)|(Brain tumor)|(Glioma brain tumor)|(Glioma Tumor)|(tumor cells)', 'Tumor', joined_data$sra_attribute.tissue, ignore.case = TRUE )
#joined_data$sra_attribute.tissue <-  gsub('Brain tumor', 'Tumor', joined_data$sra_attribute.tissue )
#joined_data$sra_attribute.tissue <-  gsub('Glioma brain tumor', 'Tumor', joined_data$sra_attribute.tissue )
#joined_data$sra_attribute.tissue <-  gsub('Glioma Tumor', 'Tumor', joined_data$sra_attribute.tissue )
#joined_data$sra_attribute.tissue <-  gsub('tumor cells', 'Tumor', joined_data$sra_attribute.tissue )

#joined_data$sra_attribute.tissue <-  gsub('Healthy Brain control', 'Brain', joined_data$sra_attribute.tissue )

  dge <- DGEList(
    counts = assay(joined_data, "counts"),
    genes = rowData(joined_data)
  )
  dge <- calcNormFactors(dge)
```


```{r}
library("ggplot2")
ggplot(as.data.frame(colData(joined_data)), aes(y = assigned_gene_prop, x = sra_attribute.Sex)) +
    geom_boxplot() +
    theme_bw(base_size = 20) +
    ylab("Assigned Gene Prop") +
    xlab("Gender Group")


```

[age|gender|tissue|cell_type|sorting_markers]

+ sra_attribute.tissue  + sra_attribute.cell_type + sra_attribute.sorting_markers  + 
[age|Sex|genotype|tissue]
```{r,   results = 'hide'}
mod <- model.matrix(~ sra_attribute.Sex +  prenatal + sra_attribute.genotype + sra_attribute.tissue,
    data = colData(joined_data)
)
colnames(mod)
```
```{r}
library("limma")

```


```{r}

vGene <- voom(dge, mod, plot = TRUE)

eb_results <- eBayes(lmFit(vGene))

de_results <- topTable(
    eb_results,
    coef = 2,
    number = nrow(joined_data),
    sort.by = "none"
)
dim(de_results)
```

```{r}
head(de_results)

```
```{r}
table(de_results$adj.P.Val < 0.05)

```


```{r}
de_results.filt <- de_results[de_results$adj.P.Val < 0.05,]  #filtering by FDR < 0.01
all_DEgenes <- de_results.filt[abs(de_results.filt$logFC)>2,] # filtering by logFC > 2
DEgenes <- rownames(all_DEgenes) 
plotMA(eb_results, coef = 2, status = rownames(eb_results$lods) %in% DEgenes, legend = FALSE,
main = "MA plot")
legend("bottomright", c("DE genes", "Not DE genes"), fill = c("red", "black"), inset = 0.01)
```
```{r}
library(EnhancedVolcano)

```


[age|Sex|genotype|tissue]
```{r}


FC <- 2
p <- 0.05
keyvals <- rep('grey75', nrow(de_results))
names(keyvals) <- rep('NS', nrow(de_results))

keyvals[which(abs(de_results$logFC  ) > FC & de_results$P.Value > p)] <- 'grey50'
names(keyvals)[which(abs(de_results$logFC) > FC & de_results$P.Value > p)] <- 'log2FoldChange'

keyvals[which(abs(de_results$logFC  ) < FC & de_results$P.Value < p)] <- 'grey25'
names(keyvals)[which(abs(de_results$logFC  ) < FC & de_results$P.Value < p)] <- '-Log10Q'

keyvals[which(de_results$logFC < -FC & de_results$P.Value < p)] <- 'blue2'
names(keyvals)[which(de_results$logFC  < -FC & de_results$P.Value < p)] <- 'Signif. down-regulated'

keyvals[which(de_results$logFC > FC & de_results$P.Value < p)] <- 'red2'
names(keyvals)[which(de_results$logFC > FC & de_results$P.Value < p)] <- 'Signif. up-regulated'

unique(keyvals)
unique(names(keyvals))

EnhancedVolcano(de_results,
  lab = de_results$gene_name,
  x = 'logFC',
  y = 'adj.P.Val',
  title = 'Volcano plot',
  pCutoff = 0.05,
  FCcutoff = 2,
  pointSize = 2.5,
  labSize = 4.5,
  colCustom = keyvals,
  colAlpha = 0.75,
  legendPosition = 'right',
  legendLabSize = 15,
  legendIconSize = 5.0,
  drawConnectors = TRUE,
  widthConnectors = 0.5,
  colConnectors = 'grey50',
  gridlines.major = TRUE,
  gridlines.minor = FALSE,
  border = 'partial')
```


```{r}
## Extraer valores de los genes de interés
exprs_heatmap <- vGene$E[rank(de_results$adj.P.Val) <= 10, ]
rownames(exprs_heatmap) <-  de_results[rownames(exprs_heatmap),'gene_name']

## Creemos una tabla con información de las muestras
## y con nombres de columnas más amigables
df <- as.data.frame(colData(joined_data)[, c("sra_attribute.Sex", "prenatal","sra_attribute.genotype","sra_attribute.tissue","study")])
colnames(df) <- c("Gender", "Prenatal", "Genotype", "Tissue", "Project")



## Hagamos un heatmap
library("pheatmap")
pheatmap(
    exprs_heatmap,
    cluster_rows = TRUE,
    cluster_cols = TRUE,
    show_rownames = FALSE,
    show_colnames = FALSE,
    annotation_col = df
)
```

Cambiar brain a prenatal bran
```{r, eval=FALSE}
## Extraer valores de los genes de interés
exprs_heatmap <- vGene$E[rank(de_results$adj.P.Val) <= 25, ]
rownames(exprs_heatmap) <-  de_results[rownames(exprs_heatmap),'gene_name']

## Creemos una tabla con información de las muestras
## y con nombres de columnas más amigables
df <- as.data.frame(colData(joined_data)[, c("sra_attribute.Sex","sra_attribute.tissue","prenatal")])
colnames(df) <- c("Gender", "Tissue",'Prenatal')

## Hagamos un heatmap
library("pheatmap")
pheatmap(
    exprs_heatmap,
    cluster_rows = TRUE,
    cluster_cols = TRUE,
    show_rownames = TRUE,
    show_colnames = FALSE,
    annotation_col = df
)



```
```{r}

## Extraer valores de los genes de interés
exprs_heatmap <- vGene$E[rank(de_results$adj.P.Val) <= 25, ]

rownames(exprs_heatmap) <-  de_results[rownames(exprs_heatmap),'gene_name']
## Creemos una tabla con información de las muestras
## y con nombres de columnas más amigables
df <- as.data.frame(colData(joined_data)[, c("sra_attribute.Sex","sra_attribute.tissue")])
colnames(df) <- c("Gender",'Tissue')

df$Tissue <-  gsub('(Brain; hippocampus)|(arcuate nucleus)', 'Brain', df$Tissue )
df$Tissue <-  gsub('(Left Brain hemisphere)|(neural progenitor cells)', 'Brain', df$Tissue )
df$Tissue <-  gsub('Brain /(frontal cortex/)', 'Brain', df$Tissue )
df$Tissue <-  gsub('GL261 tumor tissue, VEGF-A blockade,200ug mAb/ dose', 'Treated tumor', df$Tissue )

## Hagamos un heatmap
library("pheatmap")
pheatmap(
    exprs_heatmap,
    cluster_rows = TRUE,
    cluster_cols = TRUE,
    show_rownames = TRUE,
    show_colnames = FALSE,
    annotation_col = df
)
```



```{r, eval=FALSE}
complete_analysis <- function(joined_data){
  dge <- DGEList(
    counts = assay(joined_data, "counts"),
    genes = rowData(joined_data)
  )
  dge <- calcNormFactors(dge)
  
  mod <- model.matrix(~ sra_attribute.tissue +  prenatal + sra_attribute.genotype,
    data = colData(joined_data)
)
  
  vGene <- voom(dge, mod, plot = TRUE)

  vGene
}

female_data <- joined_data[,joined_data$sra_attribute.Sex == 'female']

f_vGene <- complete_analysis(female_data)

  f_eb_results <- eBayes(lmFit(f_vGene))
  
  f_de_results <- topTable(
      f_eb_results,
      coef = 2,
      number = nrow(joined_data),
      sort.by = "none"
  )
f_exprs_heatmap <- f_vGene$E[rank(f_de_results$adj.P.Val) <= 10, ]
rownames(f_exprs_heatmap) <-  f_de_results[rownames(f_exprs_heatmap),'gene_name']

## Creemos una tabla con información de las muestras
## y con nombres de columnas más amigables
df <- as.data.frame(colData(female_data)[, c( "prenatal","sra_attribute.genotype","sra_attribute.tissue","study")])
colnames(df) <- c( "Prenatal", "Genotype", "Tissue", "Project")



## Hagamos un heatmap
library("pheatmap")
pheatmap(
    f_exprs_heatmap,
    cluster_rows = TRUE,
    cluster_cols = TRUE,
    show_rownames = TRUE,
    show_colnames = FALSE,
    annotation_col = df
)
```



```{r}
complete_analysis_2 <- function(joined_data){
  dge <- DGEList(
    counts = assay(joined_data, "counts"),
    genes = rowData(joined_data)
  )
  dge <- calcNormFactors(dge)
  
  mod <- model.matrix(~ sra_attribute.Sex + sra_attribute.tissue +  prenatal + sra_attribute.genotype,
    data = colData(joined_data)
)
  
  vGene <- voom(dge, mod, plot = TRUE)

  vGene
}

female_data <- joined_data[,joined_data$sra_attribute.Sex %in% c('male','female')]

f_vGene <- complete_analysis_2(female_data)

  f_eb_results <- eBayes(lmFit(f_vGene))
  
  f_de_results <- topTable(
      f_eb_results,
      coef = 2,
      number = nrow(joined_data),
      sort.by = "none"
  )
f_exprs_heatmap <- f_vGene$E[rank(f_de_results$adj.P.Val) <= 25, ]
rownames(f_exprs_heatmap) <-  f_de_results[rownames(f_exprs_heatmap),'gene_name']

## Creemos una tabla con información de las muestras
## y con nombres de columnas más amigables
df <- as.data.frame(colData(female_data)[, c("sra_attribute.Sex", "prenatal","sra_attribute.genotype","sra_attribute.tissue","study")])
colnames(df) <- c("Sex", "Prenatal", "Genotype", "Tissue", "Project")



## Hagamos un heatmap
pheatmap(
    f_exprs_heatmap,
    cluster_rows = TRUE,
    cluster_cols = TRUE,
    show_rownames = TRUE,
    show_colnames = FALSE,
    annotation_col = df
)
```

We will choose sex based on this 6 genes Uty, Ddx3y,Kdm5d,Eif2s3y, Xist, Tsix

```{r}
sex_genes <- c('Uty','Ddx3y','Kdm5d','Eif2s3y', 'Xist', 'Tsix')
small_set <-  f_exprs_heatmap[sex_genes,]
pheatmap(
    small_set,
    cluster_rows = TRUE,
    cluster_cols = TRUE,
    show_rownames = TRUE,
    show_colnames = FALSE,
    annotation_col = df
)


```
```{r, message=FALSE}
library(dplyr)

```


```{r}
interesting_exp <- cbind(t(small_set),df)
grouped_info <- interesting_exp %>% 
  group_by(Sex) %>% 
  summarize_at(vars(sex_genes),funs(min=min, mean=mean))

grouped_info
```

```{r}
interesting_exp <- cbind(t(small_set),df)
grouped_info <- interesting_exp %>% 
  group_by(Sex) %>% 
  summarize_at(vars(sex_genes),funs(max=max, sd=sd))

grouped_info
```


```{r}
joined_data
```

```{r}
  dge <- DGEList(
    counts = assay(joined_data, "counts"),
    genes = rowData(joined_data)
  )
  dge <- calcNormFactors(dge)
  
  mod <- model.matrix(~ sra_attribute.Sex +  prenatal + sra_attribute.genotype + sra_attribute.tissue,
    data = colData(joined_data)
)
  
  vGene <- voom(dge, mod, plot = TRUE)

eb_results <- eBayes(lmFit(vGene))

de_results <- topTable(
    eb_results,
    coef = 2,
    number = nrow(joined_data),
    sort.by = "none"
)

## Extraer valores de los genes de interés
exprs_heatmap <- vGene$E[rank(de_results$adj.P.Val) <= 50, ]
rownames(exprs_heatmap) <-  de_results[rownames(exprs_heatmap),'gene_name']

## Creemos una tabla con información de las muestras
## y con nombres de columnas más amigables
df <- as.data.frame(colData(joined_data)[, c("sra_attribute.Sex", "prenatal","study")])
colnames(df) <- c("Gender", "Prenatal", "Project")



## Hagamos un heatmap
library("pheatmap")
pheatmap(
    exprs_heatmap,
    cluster_rows = TRUE,
    cluster_cols = TRUE,
    show_rownames = TRUE,
    show_colnames = FALSE,
    annotation_col = df
)
```

```{r}
raw_selection <- as_tibble(t(exprs_heatmap[sex_genes,]),rownames = "idx")
raw_selection <- raw_selection %>% 
  mutate(Predicted_sex = case_when(
    Uty < 0 & Ddx3y < 0 & Kdm5d < 0 & Eif2s3y < 0 & Xist > 0 & Tsix > 0 ~ "female",
    Uty > 0 & Ddx3y > 0 & Kdm5d > 0 & Eif2s3y > 0 & Xist < 3 & Tsix < 3 ~ "male",
    TRUE ~ "unknown"
  ))
table(raw_selection$Predicted_sex)
```

```{r}
classification <- DataFrame(raw_selection)
rownames(classification) <- classification$idx
classification
```
```{r}
test_prediction <- classification[colnames(joined_data),'Predicted_sex']==joined_data$sra_attribute.Sex
#Which ones are different from the prediction?
table(joined_data$sra_attribute.Sex[!test_prediction])

joined_data$Predicted_sex <- classification[colnames(joined_data),'Predicted_sex']
table(classification[colnames(joined_data),'Predicted_sex'])
table(joined_data$Predicted_sex)
```


```{r}
  dge <- DGEList(
    counts = assay(joined_data, "counts"),
    genes = rowData(joined_data)
  )
  dge <- calcNormFactors(dge)
  
  mod <- model.matrix(~ Predicted_sex +  prenatal + sra_attribute.genotype + sra_attribute.tissue,
    data = colData(joined_data)
)
  
  vGene <- voom(dge, mod, plot = TRUE)

eb_results <- eBayes(lmFit(vGene))

de_results <- topTable(
    eb_results,
    coef = 2,
    number = nrow(joined_data),
    sort.by = "none"
)

## Extraer valores de los genes de interés
exprs_heatmap <- vGene$E[rank(de_results$adj.P.Val) <= 50, ]
rownames(exprs_heatmap) <-  de_results[rownames(exprs_heatmap),'gene_name']

## Creemos una tabla con información de las muestras
## y con nombres de columnas más amigables
df <- as.data.frame(colData(joined_data)[, c("Predicted_sex", "prenatal","sra_attribute.tissue")])
colnames(df) <- c("Gender", "Prenatal","Tissue")



## Hagamos un heatmap
library("pheatmap")
pheatmap(
    exprs_heatmap,
    cluster_rows = TRUE,
    cluster_cols = TRUE,
    show_rownames = TRUE,
    show_colnames = FALSE,
    annotation_col = df
)
```

# Session info {.unnumbered}

```{r sessionInfo, echo=FALSE}
sessionInfo()
```
